### PR TITLE
Rename Notifications to Events and Contexts (https://github.com/aspne…

### PR DESCRIPTION
**Pull request by butlersoftware**
React to changes in notifications in aspnet/Security#47

### CODE REVIEW

File: src/AspNet.Security.OpenIdConnect.Server/project.json [hunk 1]
```diff
@@ -21,8 +21,8 @@
         },
 
         "Microsoft.Framework.WebEncoders.Core": "1.0.0-*",
-
-        "Newtonsoft.Json": "7.0.1"
+        "Newtonsoft.Json": "7.0.1",
+        "System.Dynamic.Runtime": "4.0.0"
```
Comment (PinpointTownes) [inline src/AspNet.Security.OpenIdConnect.Server/project.json]: Curious: why do we need that?

File: samples/Mvc/Mvc.Server/Providers/AuthorizationProvider.cs [hunk 2]
```diff
@@ -24,93 +24,93 @@ public sealed class AuthorizationProvider : OpenIdConnectServerProvider {
                 return;
             }
 
-            var context = notification.HttpContext.RequestServices.GetRequiredService<ApplicationContext>();
+            var applicationContext = context.HttpContext.RequestServices.GetRequiredService<ApplicationContext>();
```
Comment (PinpointTownes) [inline samples/Mvc/Mvc.Server/Providers/AuthorizationProvider.cs]: You can name that `database`, it should be clearer I guess.

File: src/AspNet.Security.OpenIdConnect.Server/Events/AuthorizationEndpointContext.cs [hunk 3]
```diff
@@ -13,15 +13,15 @@ namespace AspNet.Security.OpenIdConnect.Server {
...
     /// </summary>
-    public sealed class AuthorizationEndpointNotification : BaseNotification<OpenIdConnectServerOptions> {
+    public sealed class AuthorizationEndpointContext : BaseControlContext<OpenIdConnectServerOptions> {
         /// <summary>
         /// Creates an instance of this context
         /// </summary>
-        internal AuthorizationEndpointNotification(
-            HttpContext context,
+        internal AuthorizationEndpointContext(
+            HttpContext httpContext,
```
Comment (PinpointTownes) [inline src/AspNet.Security.OpenIdConnect.Server/Events/AuthorizationEndpointContext.cs]: I'm not sure to understand why this change is needed. Please revert all these changes :+1:

File: src/AspNet.Security.OpenIdConnect.Server/OpenIdConnectServerExtensions.cs [hunk 4]
```diff
@@ -218,38 +220,38 @@ public static class OpenIdConnectServerExtensions {
         /// Retrieves the <see cref="OpenIdConnectMessage"/> instance
         /// associated with the current request from the ASP.NET context.
         /// </summary>
-        /// <param name="notification">The ASP.NET context.</param>
```
Comment (PinpointTownes) [inline src/AspNet.Security.OpenIdConnect.Server/OpenIdConnectServerExtensions.cs]: Ah, nice catch :smile:

File: src/AspNet.Security.OpenIdConnect.Server/OpenIdConnectServerExtensions.cs [hunk 5]
```diff
@@ -218,38 +220,38 @@ public static class OpenIdConnectServerExtensions {
...
+        /// <param name="httpContext">The ASP.NET context.</param>
         /// <returns>The <see cref="OpenIdConnectMessage"/> associated with the current request.</returns>
-        public static OpenIdConnectMessage GetOpenIdConnectRequest([NotNull] this HttpContext context) {
-            return GetFeature(context).Request;
+        public static OpenIdConnectMessage GetOpenIdConnectRequest([NotNull] this HttpContext httpContext) {
```
Comment (PinpointTownes) [inline src/AspNet.Security.OpenIdConnect.Server/OpenIdConnectServerExtensions.cs]: This change is not strictly required. Please revert it.

File: src/AspNet.Security.OpenIdConnect.Server/OpenIdConnectServerHandler.cs [hunk 6]
```diff
@@ -31,21 +31,21 @@
 namespace AspNet.Security.OpenIdConnect.Server {
     internal class OpenIdConnectServerHandler : AuthenticationHandler<OpenIdConnectServerOptions> {
         protected override async Task<AuthenticationTicket> HandleAuthenticateAsync() {
-            var notification = new MatchEndpointNotification(Context, Options);
+            var matchContext = new MatchEndpointContext(Context, Options);
```
Comment (PinpointTownes) [inline src/AspNet.Security.OpenIdConnect.Server/OpenIdConnectServerHandler.cs]: `context`?

File: src/AspNet.Security.OpenIdConnect.Server/OpenIdConnectServerHandler.cs [hunk 7]
```diff
@@ -647,17 +647,17 @@ internal class OpenIdConnectServerHandler : AuthenticationHandler<OpenIdConnectS
...
-            if (notification.HandledResponse) {
+            if (authorizationContext.HandledResponse) {
                 return;
             }
 
             await ApplyAuthorizationResponseAsync(request, response);
         }
 
-        protected override async Task HandleSignOutAsync(SignOutContext context) {
+        protected override async Task HandleSignOutAsync(SignOutContext signOutContext) {
```
Comment (PinpointTownes) [inline src/AspNet.Security.OpenIdConnect.Server/OpenIdConnectServerHandler.cs]: Hum. If this collides with another context/notification, keep using notification.

File: src/AspNet.Security.OpenIdConnect.Server/project.json (hunk 1 above)
Comment (butlersoftware) [inline src/AspNet.Security.OpenIdConnect.Server/project.json]: visual studio was showing a "module should be referenced" around the JObjects error. I see now that it actually does build anyway.

Comment (PinpointTownes) [inline src/AspNet.Security.OpenIdConnect.Server/project.json]: Hum, weird. If you really need it, please move it under `dnxcore50`, since we don't want it for `dnx451`.

**Comment by PinpointTownes** [comment]
Thanks for this PR! :smile: 

Any chance you could squash your PR before I merge it?

**Comment by PinpointTownes** [comment]
Merged, thanks! :+1: 

https://github.com/aspnet-contrib/AspNet.Security.OpenIdConnect.Server/commit/a971528ef1386002a8a0cf93a7b199f823323cd4
